code/modules/antagonists/traitor/objectives/infect.dm

```diff
@@ -143,6 +143,10 @@
 		//don't take an objective target of someone who is already dead
 		fail_objective()
 
+/datum/traitor_objective/target_player/infect/proc/on_failure()
+	SIGNAL_HANDLER
+	if(objective_state == OBJECTIVE_STATE_FAILED)
+		ehms.locked = 1
```

Comment:
your booleans should be `TRUE/FALSE` defines, not 1/0
Please change this for every position in the code where you've either done `if(boolean_value == 0)` or `boolean_value = 1`
They should be compared to TRUE/FALSE instead.